my_enumerables.rb

```diff
@@ -0,0 +1,203 @@
+module Enumerable
+  def my_each
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < length
+        modified_object << (yield self[i])
+        i += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
+
+  def my_each_with_index
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < to_a.length
+        modified_object << (yield self[i], i)
+        i += 1
+      end
+      self
+    else
+      to_enum
+    end
+  end
+
+  def my_select
+    modified_object = []
+
+    i = 0
+    if block_given?
+      while i < length
+        modified_object << self[i] if (yield self[i]) == true
+        i += 1
+      end
+      modified_object
+    else
+      to_enum
+    end
+  end
+
+  def my_all?(pattern = nil)
+    true if length.zero?
+    selected_array = []
+
+    i = 0
+    if block_given?
+      selected_array = my_select { |element| yield element }
+    elsif !block_given? && pattern
+      while i < length
+        selected_array << self[i] if pattern === self[i]
+        i += 1
+      end
+    elsif !block_given? && !pattern
+      while i < length
+        selected_array << self[i] if self[i]
+        i += 1
+      end
+    end
+    selected_array.length == length
+  end
```

Comment:
`my_all?` identical and returns the same thing as ruby's [`all?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-all-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_all.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))
my_all? can also interact with range. But your one is giving me an error. Please run the following code.
```ruby
(1...3).my_all? # should return true
(1...3).my_all?(Integer) # should return true
```
So all the operation it can perform with array should be able to do with range. The answer is in the same solution that I have given.